addon/components/frost-pick-list.js

```diff
@@ -1,6 +1,125 @@
 import Ember from 'ember'
-const {Component} = Ember
+const {A, Component, copy} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import {sort} from 'ember-frost-sort'
+import layout from '../templates/components/frost-pick-list'
+import {PropTypes} from 'ember-prop-types'
 
 export default Component.extend({
-  classNames: ['frost-pick-list']
+  // == Dependencies ==========================================================
+
+  // == Properties ============================================================
+
+  classNames: ['frost-pick-list'],
+  itemsCached: A([]),
+  inputFilter: '',
+  onlySelected: false,
+  layout,
+
+  // == PropTypes =============================================================
+
+  propTypes: {
+    // options
+    listComponent: PropTypes.object.isRequired,
+    selectedItems: PropTypes.arrayOf(PropTypes.object),
+    items: PropTypes.arrayOf(PropTypes.object),
+    sortOrder: PropTypes.arrayOf(PropTypes.string),
+    sortingProperties: PropTypes.arrayOf(PropTypes.object),
+    sortOrderMax: PropTypes.number
+  },
+
+  getDefaultProps () {
+    return {
+      // options
+      items: A([]),
+      sortOrder: A([]),
+      selectedItems: A([]),
+      sortingProperties: A([]),
+      sortOrderMax: 1
+    }
+  },
+
+  // == Computed Properties ===================================================
+
+  @readOnly
+  @computed('onlySelected')
+  /**
+   * show all record or just selected
+   * @param {Boolean} onlySelected - should show only selected?
+   * @returns {Boolean} should show only selected?
+   */
+  onlySelectedComputed (onlySelected) {
+    return {
+      all: onlySelected ? 'tertiary' : 'primary',
+      selected: onlySelected ? 'primary' : 'tertiary'
+    }
+  },
+
+  @readOnly
+  @computed('items.[]', 'selectedItems.[]')
+  /**
+   * Show amount of visible data
+   * @param {Array} items - all items
+   * @param {Array} selectedItems - all selected items
+   * @returns {String} the sub title
+   */
+  subTitle (items, selectedItems) {
+    const total = this.get('itemsCached').length
+    const amtSelected = selectedItems ? selectedItems.length : 0
+
+    return `Showing ${items.length} of ${total} (${amtSelected} selected)`
+  },
+
+  // == Functions =============================================================
+
+  init () {
+    this._super(...arguments)
+    this.get('itemsCached').setObjects(copy(this.get('items')))
+    this.processItems('itemsCached')
+  },
+
+  processItems (itemsToProcess) {
```

Comment:
I think this should be a cp instead. an 'items' cp that performs the sorting. You can have another 'filteredItems' cp that does the filtering. This 'items' cp would have dependent keys of onlySelected and itemsToProcess..
Having these two functions seems like anti pattern

Reply:
I<ve started this component like this but there seem to be a problem with the `frost-table`. When I switch the `items` variable to a computed property, the pick-list works perfectly with a `frost-list` but when I put a `frost-table` I keep getting 
`TypeError: Cannot convert object to primitive value`.
the "anti pattern" was a compromise so it works with both components.

Reply:
Instead of compromising fix `frost-table`